### ISSUE TITLE
type arguments of type aliases aren't checked against their definitions, only the underlying type

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#969]
```py
class A[T: (int, str)]: ...

type BoolA[T: bool] = A[T]
BoolA[str]
BoolA[bool]

type B[T: str] = object
type C[T: int] = B[T]  # error

C[str]
B[int]
```

**Comment by DetachHead** [comment]
looks like this only happens when `enableExperimentalFeatures` is disabled and the type alias is used as a value instead of an annotation.

Code sample in [basedpyright playground](https://basedpyright.com/?pyrightVersion=1.20.0&typeCheckingMode=all&code=MYGwhgzhAECCDaAVAXNAFASwHYBcA00EOATgJQC6qAdDQFC04CeADgKbQBCA9lyAitABGPEOWgBeOEjEBiaFi7RWxYl2LoAZmBAR2WVgHMwODADdWpAgBMMVrAHIc0AO7ttxVmCuNoACzDm0GCErMxgxMbsGFAAruwaatA4vtEA-LTcvAhExLLyisqq6mhaOnqGxmYWGSIIwrx5CkoqiWjAah7AOKT0TGycSKg5YpJcggBWrF0MLOwAwoPQ2DgjA4hi0HKFre0qU930C8Ob%2Bc1Fmtq68hUm5j0c8MuNBS3FpVf6RrfVtAlcqEcSHltsVdp0DoJwqgHk8TiD0GD9qQgA)

```python
class A[T: (int, str)]: ...

type BoolA[T: bool] = A[T] # no error (false negative), didn't we already have a separate issue for this?
BoolA[str] # no error (false negative)
BoolA[bool] # no error (correct)

type B[T: str] = object
type C[T: int] = B[T]  # error (correct)

C[str] # no error (false negative)
B[int] # no error (false negative)

foo: C[str] # error (correct)
bar: B[int] # error (correct)
```
